lib/liquid/template.rb

```diff
@@ -64,11 +62,11 @@ class << self
       attr_writer :taint_mode
 
       def file_system
-        @@file_system
+        Thread.current[:liquid_file_system] ||= BlankFileSystem.new
```

Comment:
I think it would be quite surprising to set Liquid::Template.file_system and then find it returns a BlankFileSystem instance on another thread.  The point is that BlankFileSystem isn't an appropriate default, so the application would want to specify a different default file system during initialization.  Having a per-thread default seems more complicated to reason about than using `context.registers[:file_system] =` to change that state for the code that uses a different file system.